src/main/java/com/microsoft/graph/content/MSBatchResponseContent.java

```diff
@@ -104,20 +86,76 @@ public Response getResponseById(String requestId) {
 		return null;
 	}
 	
-	/*
-	 * @return responses as a string
+	/**
+	 * Get map of id and responses
+	 * 
+	 * @return responses in Map of id and response
 	 */
-	public String getResponses() {
-		return batchResponseArray != null ? batchResponseArray.toJSONString() : null;
+	public Map<String, Response> getResponses() {
+		if(batchResponseArray == null) 
+			return null;
+		Map<String, Response> responsesMap = new HashMap<>();
+		for(String id : batchRequestsHashMap.keySet()) {
+			responsesMap.put(id, getResponseById(id));
+		}
+		return responsesMap;
+	}
+	
+	/**
+	 * Get iterator over the responses
+	 * 
+	 * @return iterator for responses
+	 */
+	public Iterator<Map.Entry<String, Response>> getResponsesIterator() {
+		if(batchResponseArray == null) 
```

Comment:
can you just call getResponses method, get the map and return iterator, to avoid duplicate code?

Reply:
Updated the getResponsesIterator